storage/client.go

```diff
@@ -48,6 +49,10 @@ const (
 	userAgentHeader = "User-Agent"
 )
 
+var (
+	validStorageAccount = regexp.MustCompile("^[0-9a-z]{3,24}$")
```

Comment:
Well I suppose if we had a swagger for this it would be codified there and would be handled per normal validation logic, and if it changes in the swagger we'd get updated validation.